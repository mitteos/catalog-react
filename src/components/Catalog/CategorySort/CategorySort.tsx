import React from 'react';
import styled from "styled-components";
import {useAppDispatch, useAppSelector} from "hooks/redux";
import {productActions} from "store/features/product";

interface CategorySortProps {
    className?: string
}

export const CategorySort: React.FC<CategorySortProps> = ({className}) => {

    const {items} = useAppSelector(state => state.type)
    const {selectedType} = useAppSelector(state => state.product)
    const dispatch = useAppDispatch()

    const setTypeFilter = (id: number) => {
        if(selectedType === id) {
            dispatch(productActions.setSelectedType(null))
        } else {
            dispatch(productActions.setSelectedType(id))
        }
    }

    return (
        <Container className={className}>
            {items.map(category =>
                <Item
                    key={category.id}
                    onClick={() => setTypeFilter(category.id)}
                    $isActive={category.id === selectedType}
                >{category.name}</Item>
            )}
        </Container>
    );
};

const Container = styled.div`
  width: 100%;
  display: flex;
  gap: 11px;
  align-items: flex-start;
  flex-wrap: wrap;
`
const Item = styled.p<{$isActive: boolean}>`
  padding: 18px 14px;
  box-shadow: 0 15px 70px -11px rgba(43, 28, 1, 0.04);
  border-radius: 10px;
  font-weight: 400;
  font-size: 14px;
  line-height: 17px;
  text-align: center;
  align-self: stretch;
  align-items: center;
  max-width: 120px;
  cursor: pointer;
  transition: all .3s ease;
  @media (min-width: 900px) {
    background: ${({$isActive}) => $isActive ? "#FFC85E" : "#FFFFFF"};
    color: ${({$isActive}) => $isActive ? "#FFF" : "#3F4E65"};
    &:hover {
      transform: translateY(-5px);
      box-shadow: 0 15px 70px -11px rgba(43, 28, 1, 0.2);
    }
  }
  @media (max-width: 900px) {
    padding: 0;
    background: none;
    box-shadow: none;
    border-radius: 0;
    text-align: left;
    max-width: none;
    color: ${({$isActive}) => $isActive ? "#FFC85E" : "#3F4E65"}
  }
`
