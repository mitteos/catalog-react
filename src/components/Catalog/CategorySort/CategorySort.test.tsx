import {CategorySort} from "components/Catalog/CategorySort/CategorySort";
import {renderWithRedux} from "tests/helpers/renderWithRedux";
import {fireEvent, screen} from "@testing-library/react";
import * as ReduxHooks from "react-redux";


const mockStoreState =
{
    items: [
        {id: 1, name: "test 1"},
        {id: 2, name: "test 2"},
        {id: 3, name: "test 3"},
    ]
}

jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useSelector: jest.fn(),
}));

describe("category test", () => {
    test("select category", async () => {
        jest.spyOn(ReduxHooks, "useSelector").mockReturnValue(mockStoreState)
        renderWithRedux(<CategorySort />)
        const btn = await screen.getByTestId("category-btn-1")
        fireEvent.click(btn)
        expect(screen.getByTestId("category-sort-container")).toMatchSnapshot()
    })
})
