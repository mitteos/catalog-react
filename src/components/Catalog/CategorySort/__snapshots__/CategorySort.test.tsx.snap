// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`category test select category 1`] = `
<div
  class="sc-beqWaB kHTqqt"
  data-testid="category-sort-container"
>
  <p
    class="sc-gueYoa fjOcFQ"
    data-testid="category-btn-1"
  >
    test 1
  </p>
  <p
    class="sc-gueYoa fjOcFQ"
    data-testid="category-btn-2"
  >
    test 2
  </p>
  <p
    class="sc-gueYoa fjOcFQ"
    data-testid="category-btn-3"
  >
    test 3
  </p>
</div>
`;
